backend/src/main/java/com/mapbefine/mapbefine/topic/domain/Cluster.java

```diff
@@ -0,0 +1,35 @@
+package com.mapbefine.mapbefine.topic.domain;
+
+import com.mapbefine.mapbefine.pin.domain.Pin;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.Getter;
+
+@Getter
+public class Cluster {
+
+    private final double latitude;
+    private final double longitude;
+    private final List<Pin> pins;
+
+    private Cluster(double latitude, double longitude, List<Pin> pins) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.pins = pins;
+    }
+
+    public static Cluster from(Pin representPin, List<Pin> pins) {
+        return new Cluster(representPin.getLatitude(), representPin.getLongitude(), rearrangePins(representPin, pins));
+    }
+
+    private static List<Pin> rearrangePins(Pin representPin, List<Pin> pins) {
+        List<Pin> arrangePins = new ArrayList<>(List.of(representPin));
+
+        pins.stream()
+                .filter(pin -> representPin != pin)
+                .forEach(arrangePins::add);
```

Comment:
```suggestion
        pins.stream()
                .filter(pin -> !Objects.equals(pin, representPin))
                .forEach(arrangePins::add);
```

기존의 != 연산이 어떻게 수행되는지 이해가 잘 안되네요 .. 
이해좀 시켜주세여

Reply:
대표핀을 List 의 첫번째 핀으로 놓고, 나머지 핀들을 뒤에다가 위치시키는 로직이에요!

그래서 List.of 에서 representPin 을 넣어놓고, 그 뒤에 representPin 이 아닌 애들을 집어넣는거에요

Reply:
반 영 !

Reply:
중복되는게 문제라면 `Set`을 사용하는 방법도 있지 않을까요 ?

Reply:
아앗 제가 설명을 잘못한 것 같아요!

현재 Cluster가 가지고 있는 것이 {대표 좌표, 속해있는 핀들 리스트} 인데, 속해있는 핀들 리스트에서 첫번째 원소를 프론트에서 대표핀으로 사용하거든요, 그렇기 때문에, List 의 첫번째 원소로 대표 핀을 놓기 위해서 위와 같이 로직을 구성한겁니다.